tests/suites/helpers.function

```diff
@@ -65,29 +70,42 @@ typedef struct data_tag
 #define DISPATCH_UNSUPPORTED_SUITE      -5  /* Test suite not supported by the
                                                build */
 
+typedef enum
+{
+    PARAMFAIL_TESTSTATE_IDLE = 0,           /* No parameter failure call test */
+    PARAMFAIL_TESTSTATE_PENDING,            /* Test call to the parameter failure
+                                             * is pending */
+    PARAMFAIL_TESTSTATE_CALLED              /* The test call to the parameter
+                                             * failure function has been made */
+} paramfail_test_state_t;
+
 
 /*----------------------------------------------------------------------------*/
 /* Macros */
 
-/** Evaluate an expression and fail the test case if it is false.
+/**
+ * \brief   This macro tests the expression passed to it as a test step or
+ *          individual test in a test case.
  *
- * Failing the test means:
- * - Mark this test case as failed.
- * - Print a message identifying the failure.
- * - Jump to the \c exit label.
+ *          It allows a library function to return a value and return an error
+ *          code that can be tested.
  *
```

Comment:
OK, re-added. We should consider adding this up to TLS to align if we want this text everywhere.

Reply:
@AndrzejKurek when you update the submodule in https://github.com/ARMmbed/mbedtls/pull/2395 could you also add a commit to include this text in our version as well?

Reply:
Sure, I wrote that down on my to-do list.

Reply:
This can be done by cherry-picking [8954d0c](https://github.com/ARMmbed/mbed-crypto/commit/8954d0c274fedb54f416d65255d61fb8b3e909f8) ("Write documentation for TEST_ASSERT") and resolving the conflicts in the same way I did here, to give credit where credit is due and hopefully help avoid multiple implementation conflicts if we ever merge helpers.function between Mbed TLS and Mbed Crypto again.

Reply:
Done.